### ISSUE TITLE
Search doesn't work in TYPO3 8 LTS

### ISSUE THREAD

**Issue by webian** [franzholz/tt_products#18]
I'm in TYPO3 8 LTS and master branch of tt_products.
Search doesn't work and returns sql errors.
I guess that it's because
`\TYPO3\CMS\Frontend\ContentObject\ContentObjectRenderer::searchWhere`
uses named parameters, but I can be wrong since I'm new to this.
I'm still investigating but maybe you @franzholz have already a solution?

**Comment by webian** [comment]
Probably, after this
https://github.com/TYPO3/TYPO3.CMS/commit/d478cbe9955011c9fa50c0c1eeabaa8c0f181a63
the searchWhere() method can't be called directly cause they introduced named parameters

**Comment by webian** [comment]
As a temporary solution I took the old version of searchWhere() method and it works.
So I pasted that method, changing the name, in EXT:tt_products/model/class.tx_ttproducts_article_base.php

```
    public function customSearchWhere($searchWords, $searchFieldList, $searchTable = '') {
        if (!$searchWords) {
            return ' AND 1=1';
        }

        if (empty($searchTable)) {
            GeneralUtility::deprecationLog(
                'Parameter 3 of ContentObjectRenderer::searchWhere() is required can not be omitted anymore. Using Default connection!'
            );
            $queryBuilder = GeneralUtility::makeInstance(ConnectionPool::class)
                ->getConnectionByName(ConnectionPool::DEFAULT_CONNECTION_NAME)
                ->createQueryBuilder();
        } else {
            $queryBuilder = GeneralUtility::makeInstance(ConnectionPool::class)
                ->getQueryBuilderForTable($searchTable);
        }

        $prefixTableName = $searchTable ? $searchTable . '.' : '';

        $where = $queryBuilder->expr()->andX();
        $searchFields = explode(',', $searchFieldList);
        $searchWords = preg_split('/[ ,]/', $searchWords);
        foreach ($searchWords as $searchWord) {
            $searchWord = trim($searchWord);
            if (strlen($searchWord) < 3) {
                continue;
            }
            $searchWordConstraint = $queryBuilder->expr()->orX();
            $searchWord = $queryBuilder->escapeLikeWildcards($searchWord);
            foreach ($searchFields as $field) {
                $searchWordConstraint->add(
                    $queryBuilder->expr()->like($prefixTableName . $field, $queryBuilder->quote('%' . $searchWord . '%'))
                );
            }

            if ($searchWordConstraint->count()) {
                $where->add($searchWordConstraint);
            }
        }

        return ' AND ' . (string)$where;
    }
```

**Comment by franzholz** [comment]
In TYPO3 8.6.1 searchWhere gives a strange result:

>AND (\`product\`.\`title\` LIKE :dcValue1) OR (\`product\`.\`subtitle\` LIKE :dcValue2) OR (\`product\`.\`note\` LIKE :dcValue3)

**Comment by franzholz** [comment]
I have fixed the SEARCH now for tt_products 2.8.17

https://github.com/franzholz/tt_products/commit/25554a53b18582c034ed25132828baf87fadde7b

**Comment by webian** [comment]
Good job @franzholz ! Thank you!
